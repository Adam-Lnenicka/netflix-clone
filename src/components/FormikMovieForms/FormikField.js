import React from "react";
import { FormInput, Label } from "./styles/FormsStyles";

const FormikField = ({ title, path, myErrors, myTouched }) => {
  return (
    <div>
      <Label>{title}</Label>
      <FormInput
        placeholder={title}
        name={path}
        type="text"
        id={path}
      />

      {myErrors[path] && myTouched[path] ? <div>{myErrors[path]}</div> : null}
    </div>
  );
};

export default FormikField;
