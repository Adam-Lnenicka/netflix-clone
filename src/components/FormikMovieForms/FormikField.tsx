import React from "react";
import { Field } from "formik";

interface FieldInterface {
  title: string;
  path: string;
  myErrors: any;
  myTouched: any;
}

const FormikField = ({ title, path, myErrors, myTouched }: FieldInterface) => {
  return (
    <div>
      <div className="form__label">{title}</div>
      <Field
        placeholder={title}
        name={path}
        type="text"
        id={path}
        className="form__input"
      />

      {myErrors[path] && myTouched[path] ? (
        <div className="form__error-message">{myErrors[path]}</div>
      ) : null}
    </div>
  );
};

export default FormikField;
