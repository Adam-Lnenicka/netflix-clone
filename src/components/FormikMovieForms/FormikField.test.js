import "jsdom-global/register";
import { mount } from "enzyme";
import toJSON from "enzyme-to-json";
import React from "react";
import FormikField from "./FormikField";

describe("Formik Field", () => {
  it("create snapshot for Formik Field", () => {
    const props = {
      title: "test-title",
      path: "test-path",
      myErrors: "test-errors",
      myTouched: "test-touched",
    };

    const wrapper = mount(<FormikField {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
