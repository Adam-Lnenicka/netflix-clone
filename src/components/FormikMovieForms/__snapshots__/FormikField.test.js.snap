// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Formik Field create snapshot for Formik Field 1`] = `
<div>
  <div
    className="form__label"
  >
    test-title
  </div>
  <Field
    className="form__input"
    id="test-path"
    name="test-path"
    placeholder="test-title"
    type="text"
  />
</div>
`;
